XenModel/Actions/MultipleAction.cs

```diff
@@ -127,15 +143,26 @@ private void SubActionChanged(ActionBase sender)
             {
                 SubActionTitle = subAction.Title;
                 SubActionDescription = subAction.Description;
+                RecalculatePercentComplete();
                 OnChanged();
             }
         }
 
+        protected virtual void RecalculatePercentComplete()

```

Comment:
Now that I'm looking at it again, and apologies for not thinking of it the first time, wouldn't it be more efficient to increment the multiple action percentage by  an n-th of the percentage of the sender in the SubActionChanged handler instead of iterating though the relevant subactions or actionsWith/WithoutConnection? In which case, actionsCount could become protected in the parent class and the Recalculate method could go completely. Any hidden caveats in doing so?